### DISCUSSION TITLE
No game listed in steam server list

### DISCUSSION THREAD

**Discussion by AxelBremer** [lloesche/valheim-server-docker#2] [Q&A]
Hi!

Thanks for making this! When I run the container and wait for the server to be downloaded etc. it does show up in the steam server list but it doesn't list what game it is and in Valheim I don't see the server at all.

Would you maybe know how to fix this?

**Comment by lloesche** [comment]
Hi @AxelBremer,

the only networking related issue I've run into thus far where the server would start up and give a log message "Connected" with no indication of any errors but then not show up in-game was when the three UDP ports 2456-2458 were not publicly accessible from the Internet.

**Comment by anpross** [comment]
hi, noticed the same thing, however some of us where able to see the server and the others then joined via the steam buddy list to those. i was suspecting an issue with the central server list🤷‍♂️ - and also thanks!

**Comment by anpross** [comment]
did a quick search - maybe that helps: https://www.reddit.com/r/valheim/comments/lcn1i6/rented_servers_being_a_nightmare_to_find/

**Comment by kparkov** [comment]
This is server discovery related and probably out of scope for this repo. I am using it with a rented server without problems. Well, actually, it shows up in the game server list, but I can't add it as a favorite server in Steam. 🤷‍♂️

Thank you for this excellent initiative, @lloesche.

**Comment by lloesche** [comment]
I just did some more testing. So it seems that only when the server is properly configured (NAT/Firewall wise) it'll show up in the Steam server browser as well as in-game. Since you can see the server in Steam server browser I'm assuming everything is configured correctly on your end.

When I misconfigure my firewall settings it will start up with no error message but won't show either in-game or in Steam server browser.

However I'm seeing what you observed. In Steam server browser the `GAME` column will be empty. That said If you filter by Game `Valheim` all the servers listed have an empty `GAME` column.

I was still able to favorite the Server there and connect to it. Although going the Steam server browser route I had to enter the server password twice. Once in Steam and then again in-game.

I'll update the README.

**Reply by kparkov** [reply]
Actually, the Steam server is now showing up in Steam servers. I missed the (weird) detail that you need to add +1 to port when adding the server manually.

So, my server is running on port `2456`, but to discover it in Steam server list, I need to enter `<ip>:2457`.

**Reply by lloesche** [reply]
Oh that's good information. I'll add that to the README as well. Thanks!

**Reply by AxelBremer** [reply]
Yeah, so I do have it in the steam server list. Just trying in my local network, so there's no modem firewall trouble for now. But even with it in the steam server list I can't find it in game.

**Reply by AxelBremer** [reply]
Oh now I can't find it at all...

**Reply by lloesche** [reply]
Actually, adding manually doesn't work for me whatsoever. Like you said @kparkov it only finds the server when using `<ip>:2457`. Once added it'll show the correct server name but when trying to connect or viewing Server Info I get a window saying `0.0.0.0:0 Server: is not currently playing on any game server`.
An when I try to add it with `<ip>:2456` it will refuse to add the server and say there's no game running.

However when I filter by game, list all 5000+ available Valheim servers, sort them alphabetically and then look for it by name it shows up and can be added as a favorite. The favorite will then show `<ip>:2456` as the server IP and I can connect to it.

Early access quirks I guess.

**Reply by lloesche** [reply]
Ok so adding the server manually does work using `<ip>:2457`, but instead of `ADD THIS ADDRESS TO FAVOURITES` one needs to first press the `FIND GAMES AT THIS ADDRESS` button and then below the games list use `ADD SELECTED GAME SERVER TO FAVOURITES`.

**Reply by kparkov** [reply]
Yes @lloesche, found that out as well. I now have the server in favourites. 🤩

**Reply by lloesche** [reply]
@kparkov after some more testing I found that the favorites list is a bit flaky. Sometimes it'll show the bookmarked server entry but not show any games running on it. I had the best success by pressing `Refresh` and then immediately double click the server to connect to it. Then it'll usually work fine.

**Reply by kparkov** [reply]
@lloesche seems like there are some alpha kinks to iron out for the good people at Iron gate.

**Reply by kparkov** [reply]
Just an observation on my end: while I see more describing that flaky behaviour you are talking about @lloesche, I haven't myself. It is pretty stable - when it shows up, it does not disappear again.

**Comment by lloesche** [comment]
Updated the README with information on how to find and connect to the server.
https://github.com/lloesche/valheim-server-docker#finding-your-server

**Reply by lloesche** [reply]
@AxelBremer if you follow the steps in https://github.com/lloesche/valheim-server-docker#steam-server-favorites--lan-play
are you able to find and connect to your server? I just tested it on my local Windows machine running Docker Desktop in WSL2 and was able to find the game at `127.0.0.1:2457` and connect to the server.

**Reply by AxelBremer** [reply]
I'll check in a bit, I'm supposed to be working atm haha
